sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AllBranchesShouldNotHaveSameImplementation.cs

```diff
@@ -54,6 +60,22 @@ protected override void Initialize(SonarAnalysisContext context)
                 SyntaxKind.ElseClause);
         }
 
+        private static void Analyze(SyntaxNodeAnalysisContext context, SwitchExpressionSyntaxWrapper switchExpression)
+        {
+            var arms = switchExpression.Arms;
+            if (arms.Count < 2)
+            {
+                return;
+            }
+            var firstArm = arms[0];
+            if (switchExpression.HasDiscardPattern() &&
+                arms.Skip(1).All(arm => SyntaxFactory.AreEquivalent(arm.Expression, firstArm.Expression)))
+            {
+                var message = string.Format(SelectMessage, "switch");
+                context.ReportDiagnostic(Diagnostic.Create(rule, switchExpression.SwitchKeyword.GetLocation(), message));
```

Comment:
I just realised that we only highlight the `switch` keyword here, but the whole`switch`/`ternary`/`if` statement is highlighted in other cases. I'd prefer to have consistent behaviour across issues

Reply:
in addition, the C# implementation highlights [the whole switch statement](https://peach.sonarsource.com/issues?open=AWiU6ZJ402kLojS0BUje&projects=servuo&resolved=false&rules=csharpsquid%3AS3923), whereas the Java implementation [highlights just the keyword](https://peach.sonarsource.com/issues?open=AWjBUiGfkdi2hmIPvl9T&projects=org.elasticsearch%3Aparent&resolved=false&rules=java%3AS3923) , as the RSPEC mentions